src/lgar.cxx

```diff
@@ -1728,19 +1866,35 @@ extern void lgar_fix_dry_over_wet_wetting_fronts(double *mass_change, double* cu
 	  // now this is the wetting front that was below the dry wetting front
 	  current->psi_cm = calc_h_from_Se(Se_k, vg_a_k, vg_m_k, vg_n_k);
 
-	  struct wetting_front *current_local = *head;
+	  struct wetting_front *current_local = *head; //shouldn't be head, should be the highest to_bottom WF above the one that got deleted before the next to_bottom==FALSE one 
+    current_local = listFindFront(current->front_num - 1, *head, NULL);
+    if (listFindFront(current_local->front_num - 1, *head, NULL)!=NULL){
+      while (listFindFront(current_local->front_num - 1, *head, NULL)->to_bottom==TRUE){
+        current_local = listFindFront(current_local->front_num - 1, *head, NULL);
+        if (listFindFront(current_local->front_num - 1, *head, NULL)==NULL){
+          break;
+        }
+      }
+    }
 
-	  // update psi and theta for all wetting fronts above the current wetting front
+	  // update psi and theta for all wetting fronts above the current wetting front //is now for all WFs above the current WF until the next to_bottom == FALSE one 
 	  while (current_local->layer_num < layer_num_k) {
 	    int soil_num_k1 = soil_type[current_local->layer_num];
 	    theta_e_k   = soil_properties[soil_num_k1].theta_e;
 	    theta_r_k   = soil_properties[soil_num_k1].theta_r;
 	    vg_a_k      = soil_properties[soil_num_k1].vg_alpha_per_cm;
 	    vg_m_k      = soil_properties[soil_num_k1].vg_m;
 	    vg_n_k      = soil_properties[soil_num_k1].vg_n;
-	    double Se_l = calc_Se_from_theta(current->theta,theta_e_k,theta_r_k);
-	    current_local->psi_cm = calc_h_from_Se(Se_l, vg_a_k, vg_m_k, vg_n_k);
-	    current_local->theta = calc_theta_from_h(current->psi_cm, vg_a_k, vg_m_k, vg_n_k,theta_e_k,theta_r_k);
+
+      current_local->psi_cm = current->psi_cm;
+
+      if (current_local->next!=NULL){
+        if (isnan(current_local->psi_cm)){//in some highly unrealistic soils tested (for example, with a volumetric water content of 0.67 with a capillary head of 1e6 cm), this fxn would yield nan. This code corrects that.
+          //code has been further corrected shuch that this should never happen (fixed an error where Se_l, an old variable, was negative when it couldn't be). Nonetheless this should also redundantly correct the states if it does somehow happen.
+          current_local->psi_cm = current_local->next->psi_cm;
+        }
+      }
```

Comment:
This code is technically redundant because I did edit the code elsewhere such that psi_cm should always be a number by this point. It can be deleted, I just liked it for its redundancy.